omniscidb/QueryBuilder/QueryBuilder.cpp

```diff
@@ -2187,7 +2238,18 @@ BuilderNode BuilderNode::sort(const std::vector<BuilderSortField>& fields,
   // Sort over scan is not supported.
   auto base = node_;
   if (node_->is<Scan>()) {
-    base = proj().node();
+    // Filter out rowid column if it's not used in the sort.
+    bool uses_rowid = false;
+    for (auto& col : collation) {
```

Comment:
Does this loop look like a `find_if`?

Reply:
It appears that `find_if` option looks more verbose and probably less readable:
```
   bool uses_rowid =
        std::find_if(collation.begin(),
                     collation.end(),
                     [rowid_idx = node_->size() - 1](const& SortField field) {
                       return col.getField() == rowid_idx;
                     }) != collation.end();
```